linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1444,6 +1535,92 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
     def isPureExpression(tree: Tree)(implicit env: Env): Boolean =
       isExpressionInternal(tree, allowUnpure = false, allowSideEffects = false)
 
+    /** Test whether the given tree is an expression, or an RTLong computation
+     *  whose arguments are real expressions.
+     *
+     *  These can be directly assigned to an `Lhs`, but cannot otherwise be
+     *  used as expressions.
+     */
+    def isExpressionOrLongOpOfExpressions(tree: Tree)(implicit env: Env): Boolean = {
+      if (!isRTLongType(tree.tpe)) {
+        isExpression(tree)
+      } else {
+        // Unfortunately, there is significant duplication wrt. isExpressionInternal
+
+        def test(arg: Tree): Boolean = isExpression(arg)
+        def testAll(args: List[Tree]): Boolean = args.forall(test(_))
+
+        tree match {
+          case LongLiteral(_)                     => true
+          case VarRef(_)                          => true
+          case Transient(JSVarRef(_, mutable))    => true
+          case Transient(JSBoxedRTLongVarRef(_))  => true
+          case Transient(JSLongArraySelect(_, _)) => true
+
+          case Transient(PackLong(lo, hi)) =>
+            test(lo) && test(hi)
+          case Transient(Cast(expr, _)) =>
+            test(expr)
+
+          case tree @ UnaryOp(_, lhs) =>
+            canUnaryOpBeExpression(tree) && test(lhs)
+          case BinaryOp(_, lhs, rhs) =>
+            test(lhs) && test(rhs)
+
+          case Select(qualifier, _) =>
+            test(qualifier) && isDuplicatable(qualifier)
+          case SelectStatic(_) =>
+            true
+
+          case Apply(_, receiver, method, args) =>
+            test(receiver) && testAll(args)
+          case ApplyStatically(_, receiver, className, method, args) =>
+            test(receiver) && testAll(args)
+          case ApplyStatic(_, className, method, args) =>
+            testAll(args)
+          case ApplyDynamicImport(_, _, _, args) =>
+            testAll(args)
+          case ApplyTypedClosure(_, fun, args) =>
+            test(fun) && testAll(args)
+
+          case AsInstanceOf(expr, _) =>
+            test(expr)
+
+          case _ =>
+            false
+        }
+      }
+    }
+
+    /** Test whether, at the top level, the given tree (assumed of type `Long`)
+     *  is splittable.
+     */
+    def isSplittableLongAtTopLevel(tree: Tree)(implicit env: Env): Boolean = {
+      tree match {
+        case LongLiteral(_)                     => true
+        case VarRef(_)                          => true
+        case Transient(JSVarRef(_, mutable))    => true
+        case Transient(JSBoxedRTLongVarRef(_))  => true
+        case Transient(JSLongArraySelect(_, _)) => true
+        case Transient(PackLong(lo, hi))        => true
+        case Select(qualifier, _)               => true
```

Comment:
```suggestion
        case Transient(PackLong(lo, hi))        => true
        case _: RecordSelect                    => true
        case Select(qualifier, _)               => true
```

?

Or is `RecordSelect` special?